HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/controller/HistoryController.java

```diff
@@ -0,0 +1,105 @@
+package eu.neclab.ngsildbroker.historyquerymanager.controller;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.resteasy.reactive.RestResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.github.jsonldjava.core.Context;
+import com.github.jsonldjava.core.JsonLdProcessor;
+import eu.neclab.ngsildbroker.commons.datatypes.terms.AggrTerm;
+import eu.neclab.ngsildbroker.commons.datatypes.terms.AttrsQueryTerm;
+import eu.neclab.ngsildbroker.commons.datatypes.terms.TemporalQueryTerm;
+import eu.neclab.ngsildbroker.commons.tools.HttpUtils;
+import eu.neclab.ngsildbroker.commons.tools.QueryParser;
+import eu.neclab.ngsildbroker.historyquerymanager.service.HistoryQueryService;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.http.HttpServerRequest;
+
+@Singleton
+@Path("/ngsi-ld/v1/temporal/entities")
+public class HistoryController {
+
+	private final static Logger logger = LoggerFactory.getLogger(HistoryController.class);
+
+	@Inject
+	HistoryQueryService historyQueryService;
+	@ConfigProperty(name = "atcontext.url")
+	String atContextServerUrl;
+	@ConfigProperty(name = "scorpio.history.defaultLimit", defaultValue = "50")
+	int defaultLimit;
+	@ConfigProperty(name = "scorpio.history.maxLimit", defaultValue = "1000")
+	int maxLimit;
+	@ConfigProperty(name = "scorpio.history.lastN", defaultValue = "50")
+	int defaultLastN;
+	@ConfigProperty(name = "scorpio.history.maxLastN", defaultValue = "1000")
+	int maxLastN;
+	@ConfigProperty(name = "ngsild.corecontext", defaultValue = "https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld")
+	String coreContext;
+
+	@PostConstruct
+	public void init() {
+		JsonLdProcessor.init(coreContext);
+	}
+
+	@GET
+	public Uni<RestResponse<Object>> queryTemporalEntities(HttpServerRequest request,
+			@QueryParam(value = "limit") Integer limit, @QueryParam(value = "offset") Integer offset,
+			@QueryParam(value = "qtoken") String qToken, @QueryParam(value = "options") List<String> options,
```

Comment:
Add other query parameter here for implement aggregation support same as get API based on entityId

Reply:
done